### PR TITLE
satellite/gracefulexit: Add graceful exit completed/failed receipt verification to satellite CLI

### PR DESCRIPTION
**Pull request by ethanadams**
What: 
Add graceful exit completed/failed receipt verification to satellite CLI

Why:
We want to be able to verify a completed/failed receipt if a storage node opens a ticket related to payouts for graceful exit.

Please describe the tests:
 - Test 1:
 - Test 2:
 
Please describe the performance impact:

## Code Review Checklist (to be filled out by reviewer)
 - [x] NEW: Are there any Satellite database migrations? Are they forwards _and_ backwards compatible? 
 - [x] Does the PR describe what changes are being made?
 - [x] Does the PR describe why the changes are being made?
 - [x] Does the code follow [our style guide](https://github.com/storj/docs/blob/master/code/Style.md)?
 - [x] Does the code follow [our testing guide](https://github.com/storj/docs/blob/master/code/Testing.md)?
 - [x] Is the PR appropriately sized? (If it could be broken into smaller PRs it should be)
 - [x] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [x] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [x] Does any documentation need updating?
 - [x] Do the database access patterns make sense?

### CODE REVIEW

File: cmd/satellite/gracefulexit.go [hunk 1]
```diff
@@ -98,3 +104,46 @@ func generateGracefulExitCSV(ctx context.Context, completed bool, start time.Tim
 	}
 	return err
 }
+
+func verifyGracefulExitReceipt(ctx context.Context, identity *identity.FullIdentity, receipt string) error {
```
Comment (VinozzZ) [inline cmd/satellite/gracefulexit.go]: Maybe we should also have an input for the requested storage node id so we can check that the receipt is actually for that storage node?

Comment (ethanadams) [reply cmd/satellite/gracefulexit.go]: Added ID checks

File: cmd/satellite/gracefulexit.go [hunk 2]
```diff
@@ -98,3 +104,64 @@ func generateGracefulExitCSV(ctx context.Context, completed bool, start time.Tim
...
+		err = proto.Unmarshal(bytes, failed)
+		if err != nil {
+			return errs.Wrap(err)
+		}
+
+		err = signing.VerifyExitFailed(ctx, signee, failed)
+		if err != nil {
+			return errs.Wrap(err)
+		}
+		err = checkIDs(identity.PeerIdentity().ID, nodeID, failed.SatelliteId, failed.NodeId)
```
Comment (VinozzZ) [inline cmd/satellite/gracefulexit.go]: Maybe we could verify the nodeID before verify the signature since it's cheaper?

Comment (ethanadams) [reply cmd/satellite/gracefulexit.go]: valid point. updated

File: cmd/satellite/main.go [hunk 3]
```diff
@@ -287,6 +300,23 @@ func cmdQDiag(cmd *cobra.Command, args []string) (err error) {
...
+func cmdVerifyGracefulExitReceipt(cmd *cobra.Command, args []string) (err error) {
+	ctx, _ := process.Ctx(cmd)
+
+	identity, err := runCfg.Identity.Load()
+	if err != nil {
+		zap.S().Fatal(err)
+	}
+
+	// Check the receipt is not empty
+	nodeID, err := storj.NodeIDFromString(args[0])
```
Comment (mobyvb) [inline cmd/satellite/main.go]: if the first arg is nodeID, the usage description and minimum args needs to be updated, right?

Comment (ethanadams) [reply cmd/satellite/main.go]: Yes. Updated.

File: cmd/satellite/main.go [hunk 4]
```diff
@@ -287,6 +300,23 @@ func cmdQDiag(cmd *cobra.Command, args []string) (err error) {
...
 
+func cmdVerifyGracefulExitReceipt(cmd *cobra.Command, args []string) (err error) {
+	ctx, _ := process.Ctx(cmd)
+
+	identity, err := runCfg.Identity.Load()
+	if err != nil {
+		zap.S().Fatal(err)
+	}
+
+	// Check the receipt is not empty
```
Comment (VinozzZ) [inline cmd/satellite/main.go]: nit - the comment here is a little misleading?

Comment (ethanadams) [reply cmd/satellite/main.go]: fixed. old comment :)

**Pull request merged**